### PR TITLE
Add certification tests for ASB Queue binding

### PR DESCRIPTION
**Pull request by halspang**
# Description
This commit adds certification tests for the Azure Service Bus Queue
Input/Output binding. This utilizes the new certification framework
and performs a series of tests as described in the test plan readme.

The general purpose of these tests is to serve as an integration test
and as such requires an actual Azure Service Bus connection and a
Dapr sidecar.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Part of: #957

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by CodeMonkeyLeet — COMMENTED** [review]
LGTM! One extra thing: there is now a github workflow to exercise the certification tests on PR/cron schedule, so could you add this test to .github/workflows/certification.yml as well? 

Insert around line 77:

```yaml
         CRON_COMPONENTS=$(yq -I0 --tojson eval - << EOF
        - component: bindings.azure.servicebusqueues
          required-secrets: AzureServiceBusConnectionString
```

File: tests/certification/bindings/azure/servicebusqueues/servicebusqueue_test.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+	"github.com/dapr/components-contrib/tests/certification/flow/network"
+	"github.com/dapr/components-contrib/tests/certification/flow/sidecar"
+	"github.com/dapr/components-contrib/tests/certification/flow/simulate"
+	"github.com/dapr/components-contrib/tests/certification/flow/watcher"
+)
+
+const (
+	sidecarName   = "dapr-1"
+	applicationID = "app-1"
+	clusterName   = "servicebusqueuescertification"
```
[the remark this answers is not in the document]
Comment (berndverst) [reply tests/certification/bindings/azure/servicebusqueues/servicebusqueue_test.go:44]: I believe this originally came from the flow.DockerCompose command. It isn't needed.

[the remark this answers is not in the document]
Comment (halspang) [reply tests/certification/bindings/azure/servicebusqueues/servicebusqueue_test.go:44]: Yeah this just missed my purge of when I was playing with Phil's kafka stuff.

**Review by CodeMonkeyLeet — DISMISSED** [review]
LGTM!

Side note: if this isn't going in before 1.5 release, the go.mod will probably need to be updated again as the dependency revs against dapr get bumped, but right now everything looks up to date. We might also want to take https://github.com/dapr/components-contrib/pull/1273 to enable `/ok-to-test` against Azure component PRs to facilitate pre-merge testing, otherwise just keep an eye on the cert test runs after it goes in.

**Review by artursouza — DISMISSED** [review]

**Comment by artursouza** [comment]
@halspang Please, check build failure.

**Comment by halspang** [comment]
> @halspang Please, check build failure.

@artursouza - The errors are from a `go mod tidy` for the certification tests of kafka/rabbitmq. As it turns out, this module will have it too when it runs in the workflow.

The issue is, dapr depends on a *very* old version of prometheus. So old in fact, that the code we are depending on was deprecated in January of 2020 and has since been removed from the repo. Specifically, it's the `log` package. It's been replaced by `promlog`. Since we depend directly on dapr we should be forcing the package to the same version but go seems like it really wants to force us to the most recent version, not sure why yet but I'll look into it.

Seems like we should take a task to update dapr/dapr though.

**Comment by halspang** [comment]
@artursouza - Fixed the dependencies with some help from @pkedy. It won't fix the build failures but this one is fine. We'll have to get those in separate commits.

**Review by artursouza — APPROVED** [review]

**Pull request merged**